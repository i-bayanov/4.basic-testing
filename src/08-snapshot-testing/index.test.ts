import { generateLinkedList } from './index';

const elements = ['element-one', 2, , { fourthElement: 4 }];
const llinkedList = {
  value: 'element-one',
  next: {
    value: 2,
    next: {
      value: null,
      next: {
        value: { fourthElement: 4 },
        next: {
          value: null,
          next: null,
        },
      },
    },
  },
};

describe('generateLinkedList', () => {
  test('should generate linked list from values 1', () => {
    expect(generateLinkedList(elements)).toStrictEqual(llinkedList);
  });

  test('should generate linked list from values 2', () => {
    expect(generateLinkedList(elements)).toMatchSnapshot();
  });
});
