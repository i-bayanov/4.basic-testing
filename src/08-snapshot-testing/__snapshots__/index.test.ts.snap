// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": null,
          "value": null,
        },
        "value": {
          "fourthElement": 4,
        },
      },
      "value": null,
    },
    "value": 2,
  },
  "value": "element-one",
}
`;
